android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/service/BdsService.java

```diff
@@ -0,0 +1,151 @@
+package com.appcoins.sdk.billing.service;
+
+import android.os.AsyncTask;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.Map;
+
+public class BdsService implements Service {
+
+  private String baseUrl;
+
+  public BdsService(String baseUrl) {
+    this.baseUrl = baseUrl;
+  }
+
+  private static RequestResponse createRequest(String baseUrl, String endPoint, String httpMethod,
+      List<String> paths, Map<String, String> queries, String body) {
+    HttpURLConnection urlConnection = null;
+    try {
+      StringBuilder urlBuilder = buildUrl(baseUrl, endPoint, paths, queries);
+      URL url = new URL(urlBuilder.toString());
+      urlConnection = openUrlConnection(url, httpMethod);
+      int responseCode = urlConnection.getResponseCode();
+
+      InputStream inputStream;
+      if (responseCode >= 400) {
+        inputStream = urlConnection.getErrorStream();
+      } else {
+        inputStream = urlConnection.getInputStream();
+      }
+      if (httpMethod.equals("POST") && body != null) {
+        setPostOutput(urlConnection, body);
+      }
+      return readResponse(inputStream, responseCode);
```

Comment:
isthere a close urlConnection.disconnect(); missing?

Reply:
the finally code always execute unless there's a crash